tests/reftests/pin.test

```diff
@@ -1,7 +1,811 @@
+
+The following actions will be performed:
+=== install 1 package
+  - install nip-path ved (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed nip-path.ved
+Done.
+### opam pin add ./nip-path2
+Package nip-path2 does not exist, create as a NEW package? [y/n] y
+nip-path2 is now pinned to file://${BASEDIR}/nip-path2 (version ved)
+
+The following actions will be performed:
+=== install 1 package
+  - install nip-path2 ved (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed nip-path2.ved
+Done.
+### opam pin add $BASEDIR/nip-path3
+Package nip-path3 does not exist, create as a NEW package? [y/n] y
+nip-path3 is now pinned to file://${BASEDIR}/nip-path3 (version ved)
+
+The following actions will be performed:
+=== install 1 package
+  - install nip-path3 ved (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed nip-path3.ved
+Done.
+### : pin add with version
+### opam pin add nip-v 2
+nip-v is now pinned to version 2
+
+The following actions will be performed:
+=== install 1 package
+  - install nip-v 2 (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> retrieved nip-v.2  (file://${BASEDIR}/nip-v.2.tgz)
+-> installed nip-v.2
+Done.
+### : pin add with --dev
+### opam pin add nip2 --dev
+[nip2.1] synchronised (file://${BASEDIR}/nip2)
+nip2 is now pinned to file://${BASEDIR}/nip2 (version ved)
+
+The following actions will be performed:
+=== install 1 package
+  - install nip2 ved (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed nip2.ved
+Done.
+### opam pin add no-url --dev
+[no-url.1] synchronised (file://${BASEDIR}/no-url)
+no-url is now pinned to file://${BASEDIR}/no-url (version ved)
+
+The following actions will be performed:
+=== install 1 package
+  - install no-url ved (pinned)
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed no-url.ved
+Done.
+### : list
+### opam pin list
+nip-git.ved      git      git+file://${BASEDIR}/nip-git#master
+nip-git2.ved     git      git+file://${BASEDIR}/nip-git2#master
+nip-path.ved     rsync    file://${BASEDIR}/nip-path
+nip-path2.ved    rsync    file://${BASEDIR}/nip-path2
+nip-path3.ved    rsync    file://${BASEDIR}/nip-path3
+nip-v.2          version  2
+nip2.ved         rsync    file://${BASEDIR}/nip2
+no-url.ved       rsync    file://${BASEDIR}/no-url
+### find OPAM/targets/lib/ -name '*.t' | sort
+OPAM/targets/lib/nip-git/nip-git.ved.t
+OPAM/targets/lib/nip-git2/nip-git2.ved.t
+OPAM/targets/lib/nip-path/nip-path.ved.t
+OPAM/targets/lib/nip-path2/nip-path2.ved.t
+OPAM/targets/lib/nip-path3/nip-path3.ved.t
+OPAM/targets/lib/nip-v/nip-v.2.t
+OPAM/targets/lib/nip2/nip2.ved.t
+OPAM/targets/lib/no-url/no-url.ved.t
+### :::::::::::::::::::::::
+### :II: Pin kinds
+### :::::::::::::::::::::::
```

Comment:
Add also a test for OPAMPINKIND ?

Reply:
I don't think this is necessary but if you want go ahead